packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1521,6 +1531,16 @@ function attachSuspenseRetryListeners(finishedWork: Fiber) {
         thenable.then(retry, retry);
       }
     });
+  } else {
+    let retryTime;
+    if (suspenseState !== null) {
+      retryTime = suspenseState.retryTime;
+    } else {
+      const currentTime = requestCurrentTimeForUpdate();
+      retryTime = computeExpirationForFiber(currentTime, finishedWork, null);
+    }
+
+    markUpdateTimeFromFiberToRoot(finishedWork, retryTime);
```

Comment:
I was wrong about this part. Re-rendering doesn't work because when we re-render it will trigger the same path (`hasInvisibleParentBoundary` is true), causing the infinite loop.

Lemme think about it and I'll get back to you.